pkg/controller/controller_utils.go

```diff
@@ -824,22 +826,27 @@ func (s ActivePodsWithRanks) Swap(i, j int) {
 // Less compares two pods with corresponding ranks and returns true if the first
 // one should be preferred for deletion.
 func (s ActivePodsWithRanks) Less(i, j int) bool {
-	// 1. Unassigned < assigned
+	// 1. Delete a Specific Pod 
+	if s.Pods[i].Annotations[SpecifiedToDeleteAnnotation] != s.Pods[j].Annotations[SpecifiedToDeleteAnnotation] ||
+		(s.Pods[i].Annotations[SpecifiedToDeleteAnnotation] == s.Pods[j].Annotations[SpecifiedToDeleteAnnotation] && s.Pods[j].Annotations[SpecifiedToDeleteAnnotation] == "true") {
+		return s.Pods[i].Annotations[SpecifiedToDeleteAnnotation] == "true"
+	}
```

Comment:
looks better and more readable this way, imo
```suggestion
	// 1. Delete a Specific Pod
	podIHasDeleteAnnotation := s.Pods[i].Annotations[SpecifiedToDeleteAnnotation] == "true"
	podJHasDeleteAnnotation := s.Pods[j].Annotations[SpecifiedToDeleteAnnotation] == "true"
	annotationsAreDifferent := s.Pods[i].Annotations[SpecifiedToDeleteAnnotation] != s.Pods[j].Annotations[SpecifiedToDeleteAnnotation]

	if annotationsAreDifferent || podJHasDeleteAnnotation {
		return podIHasDeleteAnnotation
	}

```